azure/services/async/async.go

```diff
@@ -69,61 +85,83 @@ func processOngoingOperation(ctx context.Context, scope FutureScope, client Futu
 }
 
 // CreateResource implements the logic for creating a resource Asynchronously.
-func CreateResource(ctx context.Context, scope FutureScope, client Creator, spec azure.ResourceSpecGetter, serviceName string) (interface{}, error) {
+func (s *AsyncService) CreateResource(ctx context.Context, spec azure.ResourceSpecGetter, serviceName string) (interface{}, error) {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "async.Service.CreateResource")
 	defer done()
 
 	resourceName := spec.ResourceName()
 	rgName := spec.ResourceGroupName()
 
 	// Check if there is an ongoing long running operation.
-	future := scope.GetLongRunningOperationState(resourceName, serviceName)
+	future := s.Scope.GetLongRunningOperationState(resourceName, serviceName)
 	if future != nil {
-		return processOngoingOperation(ctx, scope, client, resourceName, serviceName)
+		return processOngoingOperation(ctx, s.Scope, s.Creator, resourceName, serviceName)
 	}
 
 	// No long running operation is active, so create the resource.
-	scope.V(2).Info("creating resource", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
-	result, sdkFuture, err := client.CreateOrUpdateAsync(ctx, spec)
+
+	// Check if the resource already exists. If it does and matches the spec, we can return early.
```

Comment:
> If it does and matches the spec, we can return early.

this is not always true, it is up to the `Parameters()` func to decide.

```suggestion
	// Get the resource if it already exists, and use it to construct the desired resource parameters.
```